src/App.tsx

```diff
@@ -1,14 +1,55 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { check, getTodos } from './api';
+import { Todo } from './types/Todo';
+import { TodoUser } from './types/User';
 
 export const App: React.FC = () => {
+  const [sourceData, setSourceData] = useState<Todo[] | null>(null);
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [sortBy, setSortBy] = useState('all');
+  const [userModal, setUserModal] = useState<TodoUser | null>(null);
+  const [eyeMark, setEyeMark] = useState<number>(-1);
+
+  const preRender = () => {
```

Comment:
consider renaming this function to smth like `getVisibleTodos` etc., to be more descriptive